packages/react-core/src/components/Button/Button.tsx

```diff
@@ -95,6 +95,8 @@ export interface ButtonProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'r
   innerRef?: React.Ref<any>;
   /** Adds count number to button */
   countOptions?: BadgeCountObject;
+  /** @hide Sets the role of the button. Should only be used when the button is a descendant of a menu or tablist. */
+  role?: 'menuitem' | 'tab';
```

Comment:
I'm hesitant to expose this publicly since it's a very specific use-case. I originally had just imported the button styles into MenuItemAction and applied the necessary class names to a native `<button>` element, so we could do that instead.

This could be worth exposing to the consumer once we can create some demos on when/how to properly use the prop.